packages/paste-core/components/minimizable-dialog/src/MinimizableDialogButton.tsx

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {Button} from '@twilio-paste/button';
+import {NonModalDialogDisclosurePrimitive} from '@twilio-paste/non-modal-dialog-primitive';
+import {MinimizableDialogContext} from './MinimizableDialogContext';
+import type {ButtonProps} from '@twilio-paste/button';
+
+type MinimizableDialogButtonProps = Omit<ButtonProps, 'as'> & {
+  id?: string;
+  toggle?: () => void;
+};
+
+const MinimizableDialogButton = React.forwardRef<HTMLButtonElement, MinimizableDialogButtonProps>(
+  ({children, element = 'MINIMIZABLE_DIALOG_BUTTON', ...props}, ref) => {
+    const dialog = React.useContext(MinimizableDialogContext);
+
+    return (
+      <NonModalDialogDisclosurePrimitive element={element} {...(dialog as any)} {...props} as={Button} ref={ref}>
```

Comment:
question: what causes the need for `as any` here?

Reply:
👀 nothing, just that I copied it from popover. will fix.